packages/connection-info/src/connection-secrets.ts

```diff
@@ -190,10 +181,75 @@ export function extractSecrets(connectionInfo: Readonly<ConnectionInfo>): {
   return { connectionInfo: connectionInfoWithoutSecrets, secrets };
 }
 
-function omitPropertiesWhoseValuesAreEmptyObjects<
-  T extends Record<string, Record<string, unknown>>
->(obj: T): Partial<T> {
+function omitPropertiesWhoseValuesAreEmptyObjects<T extends Document>(obj: T) {
   return Object.fromEntries(
     Object.entries(obj).filter(([, value]) => Object.keys(value).length > 0)
-  ) as Partial<T>;
+  );
+}
```

Comment:
This just removes the return type annotation and reduces type safety overall – any particular reason for that?

Reply:
I mean the return type annotation is still missing but that's not a big deal

Reply:
🤦🏽